src/likelihoods/categorical.jl

```diff
@@ -0,0 +1,19 @@
+
+"""
+    CategoricalLikelihood
+
+Categorical likelihood is to be used if we assume that the 
+uncertainity associated with the data follows a Categorical distribution.
+```math
+    p(y|f_1, f_2, \\dots, f_n) = Categorical(y | f_1, f_2, \\dots, f_n)
+```
+On calling, this would return a Categorical distribution with `f_i` 
+probability of `i` category.
+"""
+struct CategoricalLikelihood end
+
+@functor CategoricalLikelihood
```

Comment:
There are no parameters here, so no need to use `@functor`:
```suggestion
```

Reply:
If you don't define it, `functor` just falls back to https://github.com/FluxML/Functors.jl/blob/fa1a212702fe34f06b3cb18ca2fd260c9c6dcd4c/src/functor.jl#L1-L2

Reply:
Wasn't aware of this. Thank you!